inc/Engine/Optimization/RUCSS/Warmup/ResourceFetcherProcess.php

```diff
@@ -0,0 +1,124 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Optimization\RUCSS\Warmup;
+
+use WP_Rocket\Engine\Optimization\RUCSS\Database\Queries\ResourcesQuery;
+use \WP_Rocket_WP_Background_Process;
+
+class ResourceFetcherProcess extends WP_Rocket_WP_Background_Process {
+
+	/**
+	 * Background process action name.
+	 *
+	 * @var string
+	 */
+	protected $action = 'rucss_warmup_resource_fetcher';
+
+	/**
+	 * Background process prefix.
+	 *
+	 * @var string
+	 */
+
+	protected $prefix = 'rocket';
+
+	/**
+	 * ResourcesQuery instance.
+	 *
+	 * @var ResourcesQuery
+	 */
+	private $resources_query;
+
+	/**
+	 * APIClient instance.
+	 *
+	 * @var APIClient
+	 */
+	private $api_client;
+
+	/**
+	 * ResourceFetcherProcess constructor.
+	 *
+	 * @param ResourcesQuery $resources_query ResourcesQuery instance.
+	 * @param APIClient      $api_client APIClient instance.
+	 */
+	public function __construct( ResourcesQuery $resources_query, APIClient $api_client ) {
+		parent::__construct();
+
+		$this->resources_query = $resources_query;
+		$this->api_client      = $api_client;
+	}
+
+
+	/**
+	 * Do the task for each page resources.
+	 *
+	 * @param array $resource Resource array consists of url, content and type.
+	 *
+	 * @return false
+	 */
+	protected function task( $resource ) {
```

Comment:
Could this function be simplied?
Like:
1. Resource -> add or update in DB.
2. Check if content is changed -> send / resend to Warmup

Reply:
Adjusted the code and put it inside the query class